main/src/cgeo/geocaching/maps/CGeoMap.java

```diff
@@ -360,15 +354,11 @@ private void showProgress(final boolean show) {
 
         @Override
         public void handleRegularMessage(final Message msg) {
-            if (msg.what == UPDATE_PROGRESS) {
-                if (waitDialog != null) {
+            if (waitDialog != null) {
+                if (msg.what == UPDATE_PROGRESS) {
                     final int secondsElapsed = (int) ((System.currentTimeMillis() - detailProgressTime) / 1000);
-                    final int secondsRemaining;
-                    if (detailProgress > 0) {
-                        secondsRemaining = (detailTotal - detailProgress) * secondsElapsed / detailProgress;
-                    } else {
-                        secondsRemaining = (detailTotal - detailProgress) * secondsElapsed;
-                    }
+                    assert detailProgress > 0;
```

Comment:
We generally don't use Java assert statements (except for helping the null analysis in some cases). Either you _are_ sure and the code cannot ever reach the deleted else, then the assert is not needed. Or you are not sure, then the refactoring is wrong.